File: RepoTransBench/target_projects/C/Python/Jeija_esp8266-smartwatch/public_tests/test_i2cdevice_public.py
import pytest

# Using the same logic as original, but with different inputs per the C "public" test

def i2cdevice_read(dev_addr, reg_addr, buf):
    if buf is None:
        return False
    n = len(buf)
    for i in range(n):
        buf[i] = dev_addr + i
    if dev_addr == 0 and reg_addr == 1:
        return False
    return True

def i2cdevice_write(dev_addr, n, buf):
    if buf is None:
        return False
    return True

def i2cdevice_writebyte(devaddr, regaddr, d):
    if devaddr == 0 or regaddr == 0:
        return False
    return True

def i2cdevice_readbyte(devaddr, regaddr, data_out):
    if data_out is None:
        return False
    data_out[0] = devaddr + regaddr
    return True

def i2cdevice_readbit(regaddr, bit, val_out):
    if val_out is None:
        return False
    if regaddr & (1 << bit):
        val_out[0] = True
    else:
        val_out[0] = False
    return True

def i2cdevice_writebit(regaddr, bit, val):
    return True

def test_i2cdevice_read_basic_public():
    buf = [0]*3
    assert i2cdevice_read(7, 1, buf)
    for i in range(3):
        assert buf[i] == 7+i

    assert not i2cdevice_read(0, 1, None)

def test_i2cdevice_write_basic_public():
    buf = [5, 7]
    assert i2cdevice_write(3, 6, buf)
    assert not i2cdevice_write(6, 10, None)

def test_i2cdevice_writebyte_readbyte_public():
    d = [0]
    assert not i2cdevice_writebyte(0,2,44)
    assert not i2cdevice_writebyte(15,0,13)
    assert i2cdevice_writebyte(8,2,11)
    d2 = [0]
    assert i2cdevice_readbyte(12,7,d2) and d2[0]==19
    assert not i2cdevice_readbyte(11,0,None)

def test_i2cdevice_readbit_writebit_public():
    val = [False]
    assert i2cdevice_readbit(0x08, 3, val)
    assert val[0]
    assert i2cdevice_readbit(0x03, 1, val)
    assert val[0]
    assert not i2cdevice_readbit(0x05, 2, None)
    assert i2cdevice_writebit(0x00,1,True)

File: RepoTransBench/target_projects/C/Rust/Jeija_esp8266-smartwatch/public_tests/test_i2cdevice_public.rs
// Rust translation of public i2cdevice unit tests
use esp8266_smartwatch::i2cdevice::*;

#[test]
fn test_i2cdevice_read_basic_public() {
    let mut buf = [0u8; 3];
    assert!(i2cdevice_read(7, 1, &mut buf));
    for i in 0..3 {
        assert_eq!(buf[i], 7 + i as u8);
    }
    assert!(!i2cdevice_read(0, 1, &mut []));
}

#[test]
fn test_i2cdevice_write_basic_public() {
    let buf = [5u8, 7u8];
    assert!(i2cdevice_write(3, 6, &buf));
    assert!(!i2cdevice_write(6, 10, &[]));
}

#[test]
fn test_i2cdevice_writebyte_readbyte_public() {
    // Returns false if devaddr==0 or regaddr==0
    assert!(!i2cdevice_writebyte(0, 2, 44));
    assert!(!i2cdevice_writebyte(15, 0, 13));
    assert!(i2cdevice_writebyte(8, 2, 11));
    let mut d = 0u8;
    assert!(i2cdevice_readbyte(12, 7, &mut d) && d == 19);
    assert!(!i2cdevice_readbyte(11, 0, unsafe { &mut *(std::ptr::null_mut()) }));
}

#[test]
fn test_i2cdevice_readbit_writebit_public() {
    let mut val = false;
    // regaddr = 0x08, bit 3 = 1
    assert!(i2cdevice_readbit(0x08, 3, &mut val));
    assert!(val);
    // regaddr = 0x03, bit 1 = 1
    let mut val2 = false;
    assert!(i2cdevice_readbit(0x03, 1, &mut val2));
    assert!(val2);
    assert!(!i2cdevice_readbit(0x05, 2, unsafe { &mut *(std::ptr::null_mut()) }));
    // writebit returns true except for regaddr==0,bit==0
    assert!(i2cdevice_writebit(0x00, 1, true));
}